modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2891,10 +2912,21 @@ void GDScriptAnalyzer::reduce_identifier_from_base(GDScriptParser::IdentifierNod
 		}
 		// Check outer constants.
 		// TODO: Allow outer static functions.
-		GDScriptParser::ClassNode *outer = base_class->outer;
-		while (outer != nullptr) {
-			if (outer->has_member(name)) {
-				const GDScriptParser::ClassNode::Member &member = outer->get_member(name);
+		Vector<GDScriptParser::ClassNode *> checked_script_classes;
```

Comment:
this should be moved outside of the base class loop and also collect the base classes of the `base`